lib/rack/cache/metastore.rb

```diff
@@ -38,13 +38,20 @@ def lookup(request, entity_store)
       return nil if match.nil?
 
       _, res = match
-      if body = entity_store.open(res['X-Content-Digest'])
-        restore_response(res, body)
+
+      if entity_store.nil?
+        body = ['']
       else
-        # TODO the metastore referenced an entity that doesn't exist in
-        # the entitystore. we definitely want to return nil but we should
-        # also purge the entry from the meta-store when this is detected.
+        body = entity_store.open(res['X-Content-Digest'])
+        if body.nil?
+          # TODO the metastore referenced an entity that doesn't exist in
```

Comment:
The body of a rack response must respond to each and must only yield String values. Nil does not fulfill those requirements, an array with an empty string does. 

Actually I tried returning a nil body to see what happens; I'm using rack-cache to cache the responses to [RestClient](https://github.com/rest-client/rest-client) requests using [rest-client-components](https://github.com/crohr/rest-client-components). If the body of cached responses is nil, RestClient crashes. I guess other gems that assume that responses follow the rack specification would have problems as well.